pennylane/devices/tests/test_gates.py

```diff
@@ -579,169 +556,3 @@ def circuit():
 
         expected = np.abs(mat @ rnd_state) ** 2
         assert np.allclose(res, expected, atol=tol(dev.shots))
-
-
-@flaky(max_runs=10)
-class TestInverseGatesQubit:
-    """Test the device's probability vector after application of inverse of gates."""
```

Comment:
basically devices only really need to have support for the gates that don't have a native adjoint decomposition, `Adjoint(S)`, `Adjoint(T)`, `Adjoint(SX)`, and `Adjoint(SISWAP)`.  Devices don't need support for `Adjoint(PauliX)` or `Adjoint(RX)`, because those are just PauliX and RX with an inverted parameter respectively.

But yes, we have gone through and added support for the non-decomposable adjoint gates to variuos plugins.